src/include/stir/recon_buildblock/GeneralisedPrior.h

```diff
@@ -60,6 +60,17 @@ class GeneralisedPrior:
   virtual void compute_gradient(DataT& prior_gradient, 
 		   const DataT &current_estimate) =0; 
 
+  //! This should computes a single row of the Hessian
+  /*! Default implementation just call error(). This function needs to be overridden by the
+      derived class.
+      This method computes the Hessian of a particular voxel, indicated by \c coords, of the current image estimate.
+      The default method should overwrite the values in \c prior_Hessian_for_single_densel.
+   */
+  virtual Succeeded
```

Comment:
Would you like this adjusted for all methods? `compute_Hessian`, `accumulate_Hessian_times_input` and `add_multiplication_with_approximate_Hessian`??

Reply:
ok. naming of these is a bit of a mess of course. I guess we're breaking backwards compatibility already fairly hard, so what about renaming to `accumulate_Hessian_times_vector` and `accumulate_approximate_Hessian_times_vector`?

Reply:
If we will rename, I would suggest `Hessian_vector_product`. I have seen that terminology utilised around the literature a fair amount.

Reply:
I thought about this further and I dont think that these renames belong in this PR. It will impact a lot more than priors and is a little out of scope. I will create an issue that can be addressed after #901 and #902

Reply:
makes sense